client/fingerprint/consul.go

```diff
@@ -298,6 +304,72 @@ func (cfs *consulFingerprintState) grpcTLSPort(info agentconsul.Self) (string, b
 	return fmt.Sprintf("%d", int(p)), ok
 }
 
+func (cfs *consulFingerprintState) dnsPort(info agentconsul.Self) (string, bool) {
+	p, ok := info["DebugConfig"]["DNSPort"].(float64)
+	return fmt.Sprintf("%d", int(p)), ok
+}
+
+// dnsAddr fingerprints the Consul DNS address, but only if Nomad can use it
+// usefully to provide an iptables rule to a task
+func (cfs *consulFingerprintState) dnsAddr(logger hclog.Logger) func(info agentconsul.Self) (string, bool) {
+	return func(info agentconsul.Self) (string, bool) {
+
+		var listenOnEveryIP bool
+
+		dnsAddrs, ok := info["DebugConfig"]["DNSAddrs"].([]string)
+		if ok {
+			for _, dnsAddr := range dnsAddrs {
+				dnsAddr = strings.TrimPrefix(dnsAddr, "tcp://")
+				dnsAddr = strings.TrimPrefix(dnsAddr, "udp://")
+
+				parsed, err := netip.ParseAddrPort(dnsAddr)
+				if err != nil || !parsed.IsValid() {
+					logger.Warn("could not parse Consul addresses.dns config", "value", dnsAddr)
```

Comment:
are the errors from `ParseAddrPort` helpful at all?  if so, I might like to include the error string in the log line, rather than bundling these two sad conditions together.  I can just imagine myself at some point in future looking at a bug report and wondering which of these got tripped.

Reply:
it occurs to me now too that this path isn't covered in tests?

Reply:
Two bits there:
* I looked at the `IsValid()` method and it's just checking the zero value. So we can throw that away and just check the error return instead. (I guess the method exists for cases where you've constructed the `netip.Addr` manually?)
* When I re-tested the new code against a real Consul, it turns out we should be casting to `[]any` and not `[]string`. So the tests all passed but it didn't fingerprint correctly in the real world.

I've updated the code to work correctly on real Consul, barf on parsing errors as we'd expect, and added a bunch more test coverage of malformed response bodies.